### ISSUE TITLE
Multiple errors in es2015 package

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/vets-design-system-documentation#1834]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

- [Sentry logs](http://sentry.vfs.va.gov/organizations/vsp/discover/results/?field=message&name=Higher-Level+Review&query=url%3Ahttps%3A%2F%2Fwww.va.gov%2Fdecision-reviews%2Fhigher-level-review%2Frequest-higher-level-review-form-20-0996%2F%2A+level%3Aerror+message%3A%22Invalid+regular+expression%3A+unrecognized+character+after+%28%3F+SyntaxError+%3F%28generated%2Fvendors-node_modules_department-of-veterans-affairs_web-components_dist_esm-es5_va-modal_2_entry_js.entry%29%22&sort=-message&statsPeriod=90d&widths=-1) are showing numerous events with this error:

  > SyntaxError ?(generated/vendors-node_modules_department-of-veterans-affairs_web-components_dist_esm-es5_va-modal_2_entry_js.entry)
  > Invalid regular expression: unrecognized character after (?
  >
  > So it appears that it is related to the es2015 distribution build - in Sentry, this seems associated with **Safari 10.1.2 and Mobile Safari 10.0** (unsupported browser version)
  >
  > <img width="1276" alt="Sentry event showing Regular Expression error" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/136959/406d3c26-9a8e-4f7e-8ede-f092ea3bd42d">

- A second [Sentry error](http://sentry.vfs.va.gov/organizations/vsp/discover/platform-web:0d1dffdcfe834e0d91bdcfe9430c9fe6/?field=message&name=Supplemental+Claims&project=4&query=url%3Ahttps%3A%2F%2Fwww.va.gov%2Fdecision-reviews%2Fsupplemental-claim%2Ffile-supplemental-claim-form-20-0995%2F%2A+level%3Aerror+message%3A%22undefined+is+not+a+constructor+%28evaluating+%27new+Intl.NumberFormat%28j%2Cp%29%27%29+TypeError+%2Fgenerated%2Fpolyfills.entry.js+FormatDateTimePattern+FormatDateTimePattern%28generated%2Fpolyfills.entry%29%22&sort=-message&statsPeriod=7d&widths=-1) has also made an appearance:

  > `TypeError: undefined is not a constructor (evaluating 'new Intl.NumberFormat(j,p)')`
  >
  > And this error appears because some Veterans are still using Safari 9.1.3. From looking at the web components, only `va-date` and `va-memorable-date` are using this function, and only to add a leading zero to the month and day values. I think some simple string manipulation could handle this (`padStart` is also not supported by this older Safari version).
  >
  > <img width="1270" alt="Sentry event showing the INTL number format error in safari 9.1.3" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/136959/a86171ca-371f-49ce-b682-bb6817d5f2ad">

## What I expected to happen

No error

## Reproducing
- Formation version: Current
- Device: (e.g. iPhone 8, Macbook) Mac OS, iOS
- Browser: (e.g. Firefox, IE 11) Safari

Steps to reproduce:

1. Use Safari 10

## Urgency
How urgent is this request? Please select the approriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [x] This bug is not blocking any work
- [ ] Other

### Details

This bug is from an **older-unsupported version of Safari** (current is 16.5) - I'm not sure if we need to fix this bug, but we maybe should show a message that this older browser isn't supported.

**Comment by Mottie** [comment]
I've made a little progress on this:

- `Invalid regular expression: unrecognized character after (?` - is possibly related to usage of [named capturing group](https://caniuse.com/?search=Named%20capturing%20group) (requires Safari v11.1 & v11.3 in iOS) in the [`va-telephone` component](https://github.com/department-of-veterans-affairs/component-library/blob/main/packages/web-components/src/components/va-telephone/va-telephone.tsx#L74).

  ```js
  const regex = /(?<area>\d{3})(?<local>\d{3})(?<last4>\d{4})/g;
  ```

  We could fix this by using a standard regexp if this is truly the issue.
- `TypeError: undefined is not a constructor (evaluating 'new Intl.NumberFormat(j,p)'` - now has a fix in [#751](https://github.com/department-of-veterans-affairs/component-library/pull/751)

**Comment by Mottie** [comment]
Not seeing any more regular expression errors within the past week. Will re-open this ticket if either of the aforementioned errors recurs.
